rust-toolchain

```diff
@@ -0,0 +1 @@
+1.58.1
```

Comment:
Maybe instead we should use a rust-toolchain.toml file? It seems like it's the new way and we might be able to leverage it in the CI?

https://rust-lang.github.io/rustup/overrides.html#the-toolchain-file

Reply:
Done